src/Microsoft.VisualStudio.ProjectSystem.Managed/Telemetry/TelemetryEventName.cs

```diff
@@ -71,5 +71,10 @@ internal static class TelemetryEventName
         ///     Indicates that the user was notified of the suspected incremental build failure.
         /// </summary>
         public const string IncrementalBuildValidationFailureDisplayed = Prefix + "/incrementalbuild/validationfailure/displayed";
+
+        /// <summary>
+        ///     Indicates that the NuGet restore detected a cycle.
+        /// </summary>
+        public const string NuGetRestoreCycleDetected = Prefix + "/nugetrestore/cycledetected";
```

Comment:
>cycledetected [](http://example.com/codeflow?start=80&length=13)

Don't forget to get this blessed once it starts coming through.

Reply:
think it would be helpful to add additional properties in addition to cycledetected. So have the event name as : 
/nugetrestore/cycledetection and properties like vs.projectsystem.managed.nugetrestore.cycledetection.cycledetected, vs.projectsystem.managed.nugetrestore.cycledetection.durationmillis, vs.projectsystem.managed.nugetrestore.cycledetection.successfulrestorecount,
vs.projectsystem.managed.nugetrestore.cycledetection.projectid or vs.projectsystem.managed.nugetrestore.cycledetection.solutionid perhaps. Any additional items you think may be useful to help troubleshoot when we detect a cyclical restore.

Reply:
I *believe* project id and solutionid are sent automatically. Duration and succesffulrestorecount would be nice. I think succesffulrestorecount  will be unreliable as I believe we must stop as soon as a cycle is detected.